### DISCUSSION TITLE
AggregatorV3Interface.sol not found: File not found.

### DISCUSSION THREAD

**Discussion by Fahad3242** [Cyfrin/foundry-full-course-cu#2866] [Q&A] [answered]
Can anyone help me with this problem.
**Lesson 7**

```
[⠊] Compiling...
[⠒] Unable to resolve imports:
      "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol" in "/home/sheik/foundry-f23/foundry-fund-me-f23/script/PriceConverter.sol"
      "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol" in "/home/sheik/foundry-f23/foundry-fund-me-f23/script/FundMe.sol"
with remappings:
      chainlink-brownie-contracts/=/home/sheik/foundry-f23/foundry-fund-me-f23/lib/chainlink-brownie-contracts/
      forge-std/=/home/sheik/foundry-f23/foundry-fund-me-f23/lib/forge-std/src/
[⠢] Compiling 2 files with Solc 0.8.19
[⠆] Solc 0.8.19 finished in 9.57ms
Error: Compiler run failed:
Error (6275): Source "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol" not found: File not found. Searched the following locations: "/home/sheik/foundry-f23/foundry-fund-me-f23".
ParserError: Source "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol" not found: File not found. Searched the following locations: "/home/sheik/foundry-f23/foundry-fund-me-f23".
 --> script/FundMe.sol:5:1:
  |
5 | import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error (6275): Source "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol" not found: File not found. Searched the following locations: "/home/sheik/foundry-f23/foundry-fund-me-f23".
ParserError: Source "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol" not found: File not found. Searched the following locations: "/home/sheik/foundry-f23/foundry-fund-me-f23".
 --> script/PriceConverter.sol:4:1:
  |
4 | import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol"; 
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

sheik@DESKTOP-QPTDFA1:~/foundry-f23/foundry-fund-me-f23$ forge build
[⠊] Compiling...
[⠒] Unable to resolve imports:
      "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol" in "/home/sheik/foundry-f23/foundry-fund-me-f23/script/FundMe.sol"
      "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol" in "/home/sheik/foundry-f23/foundry-fund-me-f23/script/PriceConverter.sol"
with remappings:
      chainlink-brownie-contracts/=/home/sheik/foundry-f23/foundry-fund-me-f23/lib/chainlink-brownie-contracts/
      forge-std/=/home/sheik/foundry-f23/foundry-fund-me-f23/lib/forge-std/src/
[⠢] Compiling 2 files with Solc 0.8.19
[⠆] Solc 0.8.19 finished in 8.79ms
Error: Compiler run failed:
Error (6275): Source "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol" not found: File not found. Searched the following locations: "/home/sheik/foundry-f23/foundry-fund-me-f23".
ParserError: Source "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol" not found: File not found. Searched the following locations: "/home/sheik/foundry-f23/foundry-fund-me-f23".
 --> script/FundMe.sol:5:1:
  |
5 | import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error (6275): Source "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol" not found: File not found. Searched the following locations: "/home/sheik/foundry-f23/foundry-fund-me-f23".
ParserError: Source "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol" not found: File not found. Searched the following locations: "/home/sheik/foundry-f23/foundry-fund-me-f23".
 --> script/PriceConverter.sol:4:1:
  |
4 | import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol"; 
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

```

**Comment by Aashim-Limbu** [comment]
Have you installed the chainlink brownie contract . Make sure you've got it in lib folder.

![image](https://github.com/user-attachments/assets/bf8f02f7-e2da-49be-8c63-f131da144c79)

And make sure you've your .toml file configured as well

![image](https://github.com/user-attachments/assets/2fb26bd5-4359-4b7b-b3bf-3c9b55ad80be)

**Reply by Aashim-Limbu** [reply]
i also got the same issue but i got it resolved doing some changes in remapping as far as i remember . Just `alt +right click` if it takes you to inside lib folder you're good to go.👍

**Reply by Fahad3242** [reply]
I have installed chainlink brownie contracts and configured toml too already it still gave me that error

**Reply by Aashim-Limbu** [reply]
can you send me the remapping path as well as the path you're refering in the .sol code .

**Reply by Fahad3242** [reply]
The remapping path
```
remapping=["@chainlink/contracts/=lib/chainlink-brownie-contracts/contracts/"]
```
Path refered in code

```
import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol";
```

**Reply by EngrPips** [reply]
You can actually use `command + p` for Mac or `ctrl + p` for Windows and type `AggregatorV3Interface` in the search bar that appears and vscode will show the path to that file if it exists in your codebase. You can then use that path for your import.

**Reply by Fahad3242** [reply]
Yes I checked that as welll it is the same path that is written in import

**Reply by EngrPips** [reply]
Run `forge build ' and see if your code will compile

**Reply by Fahad3242** [reply]
Nope still the same error

**Reply by EngrPips** [reply]
That means you are importing from the wrong path. Show the file inside of your working directory like I show mine below.

![Screenshot 2024-11-14 at 21 39 12](https://github.com/user-attachments/assets/5ec32480-dc27-4db2-a447-df43f21ff167)

**Reply by Fahad3242** [reply]
![Capture](https://github.com/user-attachments/assets/b36ac8c6-5604-4fab-a85a-791852fae21b)

**Reply by LordAdhaar** [reply]
<img width="591" alt="Screenshot 2024-11-15 at 1 13 18 PM" src="https://github.com/user-attachments/assets/f561c976-6261-4e96-8315-49e7d28444f9">

You are referencing the src folder twice.
If you are having trouble with remappings, use direct imports like:

`import {AggregatorV3Interface} from "../lib/chainlink-brownie-contracts/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
`

**Reply by EngrPips** [reply]
Good heads-up

**Reply by Fahad3242** [reply]
same issue again 
```
[⠊] Compiling...
[⠒] Unable to resolve imports:
      "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol" in "/home/sheik/foundry-f23/foundry-fund-me-f23/script/PriceConverter.sol"
      "./lib/chainlink-brownie-contracts/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol" in "/home/sheik/foundry-f23/foundry-fund-me-f23/script/FundMe.sol"
with remappings:
      chainlink-brownie-contracts/=/home/sheik/foundry-f23/foundry-fund-me-f23/lib/chainlink-brownie-contracts/
      forge-std/=/home/sheik/foundry-f23/foundry-fund-me-f23/lib/forge-std/src/
[⠢] Compiling 2 files with Solc 0.8.19
[⠆] Solc 0.8.19 finished in 10.69ms
Error: Compiler run failed:
Error (6275): Source "script/lib/chainlink-brownie-contracts/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol" not found: File not found. Searched the following locations: "/home/sheik/foundry-f23/foundry-fund-me-f23".
ParserError: Source "script/lib/chainlink-brownie-contracts/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol" not found: File not found. Searched the following locations: "/home/sheik/foundry-f23/foundry-fund-me-f23".
 --> script/FundMe.sol:5:1:
  |
5 | import {AggregatorV3Interface} from "./lib/chainlink-brownie-contracts/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol"; 
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error (6275): Source "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol" not found: File not found. Searched the following locations: "/home/sheik/foundry-f23/foundry-fund-me-f23".
ParserError: Source "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol" not found: File not found. Searched the following locations: "/home/sheik/foundry-f23/foundry-fund-me-f23".
 --> script/PriceConverter.sol:4:1:
  |
4 | import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol"; 
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Reply by EngrPips** [reply]
> <img alt="Screenshot 2024-11-15 at 1 13 18 PM" width="591" src="https://private-user-images.githubusercontent.com/112782045/386502125-f561c976-6261-4e96-8315-49e7d28444f9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE2ODI3NDAsIm5iZiI6MTczMTY4MjQ0MCwicGF0aCI6Ii8xMTI3ODIwNDUvMzg2NTAyMTI1LWY1NjFjOTc2LTYyNjEtNGU5Ni04MzE1LTQ5ZTdkMjg0NDRmOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExNVQxNDU0MDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03YmFkZWVmZGZkNTY1MTVjYzJmZWM3MWMxODYyMmNmZDc5MDlkOGI4NTYyZGVmODM1ZDc1NGIxNWVjOTU0MjVlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.QThWQOsVUxmJRGZPL4INdGXpgClQ66zjRa8n8ZnPW5c">
> You are referencing the src folder twice. If you are having trouble with remappings, use direct imports like:
> 
> `import {AggregatorV3Interface} from "../lib/chainlink-brownie-contracts/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol"; `

@Fahad3242 use the path that @LordAdhaar suggested, and let's see if the issue persists

**Reply by LordAdhaar** [reply]
![image](https://github.com/user-attachments/assets/d457028f-901d-4d6e-a9d4-4ec0e62b69f4)

You are using single dot(./) that is used to reference the current directory
use double dots (../) to move up one folder or go back one directory.

**Reply by EngrPips** [reply]
whoop

**Reply by LordAdhaar** [reply]
Indeed

**Comment by EngrPips** [comment]
Good job here

**Reply by Fahad3242** [reply]
![image](https://github.com/user-attachments/assets/8b1fe33c-e2a7-483b-82a3-8bdddf8d2b51)
still didn't work (underlined red too now)

**Reply by Fahad3242** [reply]
Is this folder supposed to be empty or populated with the chainlink contracts?
It is the installation path written in vscode terminal
![Uploading image.png…]()

**Reply by EngrPips** [reply]
@Fahad3242, Your second image is not showing for me.

**Reply by EngrPips** [reply]
Please open the `AggregatorV3Interface` and show your `vscode` as I show mine below.

![Screenshot 2024-11-14 at 21 39 12](https://github.com/user-attachments/assets/f814bda7-7306-4690-87e1-3f9b3be5f967)

**Reply by Fahad3242** [reply]
![image](https://github.com/user-attachments/assets/8b78148a-fa5c-4f5c-918e-ef15b7711faf)

**Reply by Fahad3242** [reply]
![image](https://github.com/user-attachments/assets/c4d347d4-e3ef-4676-b08f-931d79e190a0)

**Reply by EngrPips** [reply]
try the below and lets see if it will work

```js
import {AggregatorV3Interface} from "lib/chainlink-brownie-contracts/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol"
```

**Reply by Fahad3242** [reply]
didnt work

**Reply by EngrPips** [reply]
tough, show me.

**Reply by LordAdhaar** [reply]
`import {AggregatorV3Interface} from "../lib/chainlink-brownie-contracts/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";`

Is this also not working?
Are you able to command/ctrl + click on the import and go to that file?

**Reply by Fahad3242** [reply]
@LordAdhaar @EngrPips 
![image](https://github.com/user-attachments/assets/43a21fd5-4d49-4732-bf35-4f9849968264)

**Reply by LordAdhaar** [reply]
Can you send the screenshot of FundMe.sol

**Reply by Fahad3242** [reply]
There are two AggregatorV3interface one is in
![image](https://github.com/user-attachments/assets/0e39e3f6-7e20-4eec-a624-7e8bad17c8a0)
the other one is in
![image](https://github.com/user-attachments/assets/fffbfd24-ea41-4082-9845-4846bfcf3f33)

**Reply by LordAdhaar** [reply]
Yes I see. But can you once send the FundMe.sol file or make your repository public so I can run it own my system and see what is the error. But send the screenshot of FundMe.sol file once

**Reply by Fahad3242** [reply]
@LordAdhaar  FundMe.sol
![image](https://github.com/user-attachments/assets/8f37d0a8-35ae-4a1b-a7e4-dfc401481442)

**Reply by LordAdhaar** [reply]
I think in the PriceConverter.s.sol you are using the wrong import statement. Use this once in your FundMe.sol and PriceConverter.sol, both files 

`import {AggregatorV3Interface} from "../lib/chainlink-brownie-contracts/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";`

**Reply by LordAdhaar** [reply, marked as answer]
yeah, see the issue is in PriceConverter.sol, FundMe.sol is working fine.

`import {AggregatorV3Interface} from "../lib/chainlink-brownie-contracts/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";`

Use the above in PriceCoverter.sol and let me know

**Reply by Fahad3242** [reply]
@LordAdhaar FInally!! PriceConverter.sol had the wrong import

**Reply by Fahad3242** [reply]
Thanks for the support @LordAdhaar @EngrPips

**Reply by LordAdhaar** [reply]
Glad you figured it out. 
Tip for the future, if you are having issues with imports. If you ctrl + click on the import in {}, it will take you to the file. If not there is an issue there. 

Also, your terminal exactly tells where the issue is, so pay close attention to it.

<img width="983" alt="Screenshot 2024-11-16 at 3 59 54 PM" src="https://github.com/user-attachments/assets/68177a1e-996d-483d-b55d-79e54935c701">

**Reply by EngrPips** [reply]
Great job.

**Reply by Fahad3242** [reply]
@LordAdhaar noted!

**Comment by goofylawz** [comment]
@LordAdhaar & @EngrPips, I'm facing the same issue and somehow i'm unable to resolve this following the steps you have outlined above. I'm basically received the exact same errors

![image](https://github.com/user-attachments/assets/b02f8cb3-9be7-4192-9335-5e67e65cb3e5)

Here's my FundMe.sol file:
![image](https://github.com/user-attachments/assets/602737a4-70e2-426a-ba53-073fc98f4988)

For Price Converter, I replaced the import text with what you provided above:
![image](https://github.com/user-attachments/assets/d5d57424-aec5-4606-8957-f5daf39dc8f4)

And this is the AggregatorV3Interface.sol:

![image](https://github.com/user-attachments/assets/d1d8b2ee-a93c-47a8-a6c9-ec2691daf67a)

I'm also able to navigate with cmd+ctrl+click to the right directory, if im doing it correctly.

Looking at this, do you have an idea what I might be doing wrong here?

**Reply by EngrPips** [reply]
Why is the error still pointing to `@chainlink` when you clearly did not use the `@chainlink` tag in your import, maybe try `forge clean`, then delete your `cache` folder if you have any, then retry.

**Reply by goofylawz** [reply]
![image](https://github.com/user-attachments/assets/60e4cd19-a1fa-4306-811f-7217ba008efc)

Receiving the same error.

**Reply by goofylawz** [reply]
I think I used ```@chainlink``` the first time that I tried to do ```forge build```, but then I followed the steps outlined above by you guys.

**Reply by EngrPips** [reply]
That error said you are importing with `@chainlink/contracts/src/v0.8/shared/interface/AggregatorV3Interface.sol`. Is that correct?

**Reply by goofylawz** [reply]
The first time I went with what's provided in the course and what is in the ```Remix Fund Me``` files that I downloaded.
<img width="891" alt="image" src="https://github.com/user-attachments/assets/bdad8a94-cb79-4e42-ba04-bed4b172d1b8">

Then I tested with some of the solutions shared in the comments above:
![image](https://github.com/user-attachments/assets/56229f1d-b44d-4859-bdbe-afb776e6c9dd)

With both, i'm receiving that error.
Here's my FundMe.sol:
![image](https://github.com/user-attachments/assets/b971f0a1-334b-46f1-bba2-928e33602773)

PriceConverter.sol:
![image](https://github.com/user-attachments/assets/d4177387-b6cf-4de6-b563-d37cdea539eb)

This is what i have as set in the toml file as a remapping:
![image](https://github.com/user-attachments/assets/32bf7538-ffcb-4f18-9256-5170ce9bfda6)

**Reply by goofylawz** [reply]
Hope that makes sense.

**Reply by EngrPips** [reply]
I am unsure why it is still pointing to your previous import command. Did you run `forge clean`?

**Reply by goofylawz** [reply]
Yes and I deleted the cache folder.

**Reply by EngrPips** [reply]
Tough.

**Reply by EngrPips** [reply]
Did you save your files after you made the changes?

**Reply by goofylawz** [reply]
Indeed.

**Reply by EngrPips** [reply]
Hard to tell what is wrong. Seem like you did everything correctly.

**Reply by LordAdhaar** [reply]
Is it still pointing to the @chainlink import? If forge clean is not working, what you can do is make a new project and copy the code. That way you have a new setup and it is only two files so it won't take a lot of time to copy. You can try it once and let us know

**Reply by EngrPips** [reply]
Interesting

**Reply by goofylawz** [reply]
@LordAdhaar @EngrPips, I think that did the trick:
![image](https://github.com/user-attachments/assets/04b88189-eecf-479e-9143-5a98fb9e22ce)

**Reply by EngrPips** [reply]
You are a genius, sir!

**Reply by goofylawz** [reply]
Thank you very much sirs

**Reply by LordAdhaar** [reply]
I don't get it. So did it work or not? If it did work, what was the issue?

**Reply by goofylawz** [reply]
@LordAdhaar Interestingly, i'm receiving that now that i'm trying to build tests
![image](https://github.com/user-attachments/assets/b20024b2-e57c-4b0b-b6e9-b05137b53a80)

![image](https://github.com/user-attachments/assets/eb084327-c095-4709-9bdd-43143ecd8452)

![image](https://github.com/user-attachments/assets/9b7174fd-9bac-45bd-b14a-9edf9a933914)
it again doesn't recognize the path.

I did the following commands in the terminal:
```clear```
```forge clean```
```forge build```
```forge test -vv```

**Reply by LordAdhaar** [reply]
You don't need to put FundMe before fundMe in the setUp function, you have already declared the variable above

**Reply by LordAdhaar** [reply]
Also, how are your eyes functioning, I would recommend switching to a darker theme in vscode.

**Reply by goofylawz** [reply]
I'm getting two warning and 1 error, right now:
![image](https://github.com/user-attachments/assets/d0ec7154-aecd-45f5-ae7c-b6ae3b70de04)
Before i received that last error, I received a statement that it ```can't find the test file```

**Reply by LordAdhaar** [reply]
Is it working now?

**Reply by goofylawz** [reply]
Nope

**Reply by LordAdhaar** [reply]
Did you add the pragma solidity ^0.8.19. What the warning is saying?

**Reply by goofylawz** [reply]
Yes

**Reply by goofylawz** [reply]
and i'm still seeing this error:
```
Error (2904): Declaration "FundMe" not found in "src/FundMe.sol" (referenced as "../src/FundMe.sol").
 --> test/FundMeTest.t.sol:6:1:
  |
6 | import {FundMe} from "../src/FundMe.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ````

**Reply by LordAdhaar** [reply]
Weird. I'm also confused.

**Reply by LordAdhaar** [reply]
If you cmd+click on {FundMe} does it take you to FundMe.sol?

**Reply by EngrPips** [reply]
> and i'm still seeing this error:
> 
> ```
> Error (2904): Declaration "FundMe" not found in "src/FundMe.sol" (referenced as "../src/FundMe.sol").
>  --> test/FundMeTest.t.sol:6:1:
>   |
> 6 | import {FundMe} from "../src/FundMe.sol";
>   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> ```

Did you figure it out?

**Reply by goofylawz** [reply]
> If you cmd+click on {FundMe} does it take you to FundMe.sol?

Yes. I still haven't figured it out, I tried for a few hours yesterday and I was like f*** it, I need to get my eyes off this code. I'm going to work on this today with some fresher perspective.

Here's my test code that I got from Patrick's course:
![image](https://github.com/user-attachments/assets/71eb6979-c5c0-4351-82cd-04ac9575ead0)

When i click both fundMe.sol and test.sol - it opens the file correctly for me. So, i'm still stuck trying to figure this out.

**Reply by EngrPips** [reply]
Please send your `DeployFundMe.s.sol` file.

**Reply by goofylawz** [reply]
Phind Ai suggested that I add this as a remapping:
```
[remappings]
"forge-std/=lib/forge-std/src/"
"chainlink-brownie-contracts/=lib/chainlink-brownie-contracts/"
```
However, when I add this the error changes to:
<img width="686" alt="image" src="https://github.com/user-attachments/assets/75abebed-ca53-4e2f-b68f-e315b63ef5cb">

So, it's opening a different can of worms.

**Reply by goofylawz** [reply]
> Please send your `DeployFundMe.s.sol` file.

<img width="424" alt="image" src="https://github.com/user-attachments/assets/3557cc83-8833-4c3d-8643-4c246af4fa41">

**Reply by goofylawz** [reply]
Btw, my terminal is zsh instead of bash as in Patrick's course (it defaulted to this when I started), not sure if that makes any difference.

**Reply by EngrPips** [reply]
What is the `Counter` stuff causing this error?

**Reply by goofylawz** [reply]
the same error with the FundMe.sol shows when I try to use ```forge script script/DeployFundMe.s.sol```

**Reply by EngrPips** [reply]
I have been using `zsh` all along. It doesn't make a difference that will make your project not work.

**Reply by EngrPips** [reply]
Do you have a `Counter` file in your codebase.

**Reply by goofylawz** [reply]
> What is the `Counter` stuff causing this error?

I tried Phind's AI suggestion to add this to foundry.toml 
```[remappings]
"forge-std/=lib/forge-std/src/"
"chainlink-brownie-contracts/=lib/chainlink-brownie-contracts/"
```
Which cause that counter.sol error so I removed it.

**Reply by goofylawz** [reply]
> Do you have a `Counter` file in your codebase.

It appeared with the remappings thing that I did but i removed it just now because it is not something I need

**Reply by EngrPips** [reply]
So, what is the current error you are getting.

**Reply by goofylawz** [reply]
Same stuff:
![image](https://github.com/user-attachments/assets/35c4726a-fb78-48a3-96db-656555c65eaf)

**Reply by goofylawz** [reply]
I think i found it, testing

**Reply by EngrPips** [reply]
> I think i found it, testing

You solved the problem?

**Reply by goofylawz** [reply]
> > I think i found it, testing
> 
> You solved the problem?

Yeah, so my Auto-Save in Visual Studio Code was not enabled, hence I wasn't saving with "Ctrl+S". I though that with forge build it should be taking my changes into account but that wasn't the case. Right now, the previous error regarding the ```import {FundMe} from "../src/FundMe.sol"``` is non-existent. 

![image](https://github.com/user-attachments/assets/26862fb3-aace-4520-a103-52ad197bf102)
You can see that here my test compile, with some "warnings" that I'll figure out at some point.

**Reply by EngrPips** [reply]
> Did you save your files after you made the changes?

I said it here

**Reply by goofylawz** [reply]
> > Did you save your files after you made the changes?
> 
> I said it here

I thought I did.. It's just me being retarded.. Apologies if I've lost your time, sers! It was definitely helpful and learned a lot while debugging this.

**Reply by EngrPips** [reply]
It is fine, I am glad you figured it out.

**Reply by goofylawz** [reply]
> It is fine, I am glad you figured it out.

Thank you very much once again!

**Reply by LordAdhaar** [reply]
Why do you you use Autosave when you can "cmd+S". Just curious.

**Reply by EngrPips** [reply]
> Why do you you use Autosave when you can "cmd+S". Just curious.

Autosave saves you the stress of having to press `cmd + s` after every change. I use autosave myself.

**Reply by goofylawz** [reply]
> Why do you you use Autosave when you can "cmd+S". Just curious.

I'm going to use cmd+S, it's just that I thought it auto-saves because i don't have the habit to do it manually. But now that i know, i will build that habit.

**Reply by LordAdhaar** [reply]
Nice

**Reply by EngrPips** [reply]
Indeed
